src/plugin/htmlparsefilter-naivebayes/src/java/org/apache/nutch/htmlparsefilter/naivebayes/NaiveBayesHTMLParseFilter.java

```diff
@@ -0,0 +1,214 @@
+      LOG.error("Error occured while classifying:: " + text);
+
+    }
+
+    return false;
+  }
+
+  public boolean filterUrl(String url) {
+
+    return containsWord(url, wordlist);
+
+  }
+
+  public boolean classify(String text) throws IOException {
+
+    // if classified as relevent "1" then return true
+    if (NaiveBayesClassifier.classify(text).equals("1"))
+      return true;
+    return false;
+  }
+
+  public void train() throws Exception {
+    // check if the model file exists, if it does then don't train
+    if (!FileSystem.get(conf).exists(new Path("model"))) {
+      LOG.info("Training the Naive Bayes Model");
+      NaiveBayesClassifier.createModel(inputFilePath);
+    } else {
+      LOG.info("Model already exists. Skipping training.");
+    }
+  }
+
+  public boolean containsWord(String url, ArrayList<String> wordlist) {
+    for (String word : wordlist) {
+      if (url.contains(word)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  public void setConf(Configuration conf) {
+    this.conf = conf;
+    inputFilePath = conf.get(TRAINFILE_MODELFILTER);
+    dictionaryFile = conf.get(DICTFILE_MODELFILTER);
+    if (inputFilePath == null || inputFilePath.trim().length() == 0
+        || dictionaryFile == null || dictionaryFile.trim().length() == 0) {
+      String message = "Model URLFilter: trainfile or wordlist not set in the urlfilter.model.trainfile or urlfilter.model.wordlist";
+      if (LOG.isErrorEnabled()) {
+        LOG.error(message);
+      }
+      throw new IllegalArgumentException(message);
+    }
+    BufferedReader br = null;
+
+    try {
+
+      String CurrentLine;
+
+      Reader reader = conf.getConfResourceAsReader(dictionaryFile);
+      br = new BufferedReader(reader);
+      while ((CurrentLine = br.readLine()) != null) {
+        wordlist.add(CurrentLine);
+      }
+
+    } catch (IOException e) {
+      LOG.error("Error occured while reading the wordlist");
+
+    } finally {
+      try {
+        if (br != null)
+          br.close();
+      } catch (IOException ex) {
+        ex.printStackTrace();
+      }
+    }
+
+    try {
+
+      train();
+    } catch (Exception e) {
+      // TODO Auto-generated catch block
+      LOG.error("Error occured while training");
+      e.getStackTrace();
+
+    }
+
+  }
+
+  public Configuration getConf() {
+    return this.conf;
+  }
+
+  @Override
+  public ParseResult filter(Content content, ParseResult parseResult,
+      HTMLMetaTags metaTags, DocumentFragment doc) {
+
+    Parse parse = parseResult.get(content.getUrl());
+
+    String url = content.getBaseUrl();
+    ArrayList<Outlink> tempOutlinks = new ArrayList<Outlink>();
+    String title = parse.getData().getTitle();
+    ParseStatus status = parse.getData().getStatus();
+    String text = parse.getText();
+
+    if (!filterParse(text)) { // kick in the second tier
+      // if parent page found
+      // irrelevent
+      LOG.info("ModelURLFilter: Page found irrelevent:: " + url);
```

Comment:
this isn't ModelURLFilter, it's now naive bayes htmlparser filter, please update.